service/README.rst

```diff
@@ -1,84 +1,92 @@
-that are managed by the GEOPM service are restored to the value they
-had prior to the client opening a write session.
-
-The ``geopmd`` daemon is started by the GEOPM systemd service and uses
-D-Bus for communication with client processes and for administrator
-configuration.  The GEOPM service is used to extend the
-``geopm::PlatformIO`` features of libgeopm and libgeopmpolicy.  The
-PlatformIO interface of GEOPM is extensible through IOGroup plugins
-and the GEOPM service enables the ServiceIOGroup.  The ServiceIOGroup
-is implemented to provide the requirements of the IOGroup class by
-interfacing with ``geopmd`` over the D-Bus interface.  The
-ServiceIOGroup plugin is loaded first by PlatformIO in libgeopm and
-libgeopmpolicy, and is not loaded by the PlatformIO implementation in
-libgeopmd.  For a libgeopm or libgeopmpolicy user, any IOGroup other
-than the ServiceIOGroup that provides a requested signal or control
-will be used.  The ServiceIOGroup will only be used when a user
-requests a signal or control that cannot be provided by any of the
-native IOGroup plugins loaded by the unprivileged user process.  The
-geopmd process loads PlatformIO through libgeopmd as a root user which
-has permissions to load all signals and controls from all IOGroups but
-does not load the ProfileIOGroup that provides signals derived from
-the user application.  The PlatformIO loaded by geopmd provides the
-implementation for the GEOPM service D-Bus interfaces.
-
-Architecture Diagram
---------------------
+The GEOPM Service provides a user-level interface to read telemetry
+and configure settings of heterogeneous hardware platforms. Linux
+system administrators may manage permissions for user access to
+telemetry and configuration at a fine granularity.
+
+Clients use the GEOPM Service DBus interface to interact with the
+service.  The GEOPM Service package provides access to the DBus
+interfaces from the command line, or programatically with library
+interfaces for C, C++ and Python.
+
+The service can support many simultaneous client sessions that make
+measurements, but only clients from within one Linux process session
+are granted write permission to configure hardware control values at
+any time.  When a client process session is granted write access it
+will retain that permission until the Linux session process leader of
+that client process terminates.  When the process session leader
+terminates all hardware settings that are managed by the GEOPM Service
+are restored to the values they had prior to the first client write
+request.  See `setsid(2) <https://man7.org/linux/man-pages/man2/setsid.2.html>`_
+manual for more information about the Linux session leader processs.
 
 
-.. image:: https://geopm.github.io/images/geopm-service-diagram.svg
-   :target: https://geopm.github.io/pdf/geopm-service-diagram.pdf
-   :alt:
+*
+  `Overview slides <https://geopm.github.io/pdf/geopm-service.pdf>`_
```

Comment:
I think it was nice that the diagram used to be here, but maybe you removed it so that we don't have to worry about it being overly specific and going out of date over time. Maybe note that these slides include a diagram of the architecture at the time of initial design.

Reply:
I put it back.  A little overzealous with the chisel.